src/helpers/blood-pressure-data-providers/device-blood-pressure-data-provider.ts

```diff
@@ -0,0 +1,39 @@
+import { DeviceDataNamespace, DeviceDataPoint } from "@careevolution/mydatahelps-js";
+import queryAllDeviceData from "../daily-data-providers/query-all-device-data";
+import { BloodPressureDataPoint } from ".";
+import { parseISO, startOfDay } from "date-fns";
+
+export default async function (namespace: DeviceDataNamespace, bloodPressureSystolic: string, bloodPressureDiastolic: string): Promise<BloodPressureDataPoint[]> {
+
+    function collateResults(dataPoints: DeviceDataPoint[]): BloodPressureDataPoint[] {
+        let bpDataPointsByObservationDateTime: Map<string, BloodPressureDataPoint> = new Map<string, BloodPressureDataPoint>();
+
+        dataPoints.forEach((dataPoint) => {
+            if (dataPoint.observationDate && !isNaN(Number(dataPoint.value))) {
+                var bpDate = parseISO(dataPoint.observationDate);
+                var bpDateIdentifier = bpDate.toDateString();
```

Comment:
Curious why not just use dataPoint.observationDate since it's already a string?

Reply:
Ah I see...it's just using the date part...I think that is not quite right because you could have multiple BPs on the same day.  We should average the BPs on the same day I think

Reply:
(or maybe i'm misreading how it works)

Reply:
Using dataPoint.observationDate was a good call. Updated